robot/client/client.go

```diff
@@ -687,17 +691,21 @@ func (rc *RobotClient) Refresh(ctx context.Context) (err error) {
 
 func (rc *RobotClient) updateResources(ctx context.Context) error {
 	// call metadata service.
-	names, rpcAPIs, err := rc.resources(ctx)
-	// only return if it is not unimplemented - means a bigger error came up
-	if err != nil && status.Code(err) != codes.Unimplemented {
-		return err
-	}
-	if err == nil {
-		rc.resourceNames = make([]resource.Name, 0, len(names))
-		rc.resourceNames = append(rc.resourceNames, names...)
-		rc.resourceRPCAPIs = rpcAPIs
+	var names []resource.Name
+	var rpcAPIs []resource.RPCAPI
+	var err error
+
+	if !rc.subtypesUnimplemented {
```

Comment:
Guarding `resources` seems slightly incorrect to me. Don't we want to guard the call to `rc.client.ResourceRPCSubtypes` _in_ `resources`? Won't what you have now fail to get any resource names from the micro RDK?

Reply:
I was debating this as well, having the set and check at different parts of the call stack.

adding it here felt cleaner after trying to do the same within `resources`, but I don't have strong feelings either way.

As for the last question, I don't believe so, since this flag doesn't activate until after the first time you receive an unimplemented response. I'm adding a test now for the implemented case.

Reply:
we do want to be able to keep hitting `ResourceNames` as part of client refresh (which usually happens every 10s), in case anything changes on the micro-rdk